### PR TITLE
Edit price endpoint

### PR DESCRIPTION
**Pull request by joycrwu**
## Overview

Implemented a new endpoint to edit the price of an existing item to give a discount. 

## Changes Made

Changed the post model to have an original price and an altered price. Changed some test cases to fit this new model, changed some other functions in the Post Service to fit this new model as well. 

## Test Coverage

Manual testing with Postman, will write unit test cases later and make a separate PR.

### CODE REVIEW

**Review by mateow99 — COMMENTED** [review]
Are there any test cases for this new route?

**Comment by joycrwu** [comment]
as of rn no, but i was gonna make another PR with them

File: src/types/ApiRequests.ts [hunk 1]
```diff
@@ -96,4 +100,18 @@ export interface CreateRequestRequest {
...
+// NOTIFICATION
+export interface ExpoPushMessage {
+    to: string[];
+    sound: 'default' | null;
+    title: string;
+    body: string;
+    data: JSON;
+}
+
+// export interface SaveTokenRequest {
```
Comment (snajima) [inline src/types/ApiRequests.ts:114]: remove commented code

File: src/types/ApiResponses.ts [hunk 2]
```diff
@@ -122,4 +127,28 @@ export interface GetRequestsResponse {
 
 export interface GetRequestResponse {
     request: Request;
+}
+
+// NOTIFICATIONS
+
+export interface PushTicketData {
+    status : "error" | "ok",
```
Comment (snajima) [inline src/types/ApiResponses.ts:135]: colon indicates type, "error" nor "ok" is a type

u can either make this a string or enum

File: src/types/ApiResponses.ts [hunk 3]
```diff
@@ -122,4 +127,28 @@ export interface GetRequestsResponse {
...
+    id : string,
```
Comment (snajima) [inline src/types/ApiResponses.ts:136]: make formatting consistent, colon should not be preceded with a space

**Review by snajima — CHANGES_REQUESTED** [review]
looks good! let's just make one quick test case before merging this

File: src/api/controllers/NotifController.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    constructor(notifService: NotifService) {
+      this.notifService = notifService;
+    }
+
+    @Post()
+    async sendPost( notifRequest : ExpoPushMessage ) {
+        return this.notifService.sendNotifs(notifRequest, {});
+    }
+}
```
Comment (snajima) [inline src/api/controllers/NotifController.ts:17]: why are notification changes also included in this PR?

File: src/services/NotifService.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                data: notif.data
+            })
+        })
+
+        this.sendNotifChunks(notifs, expoServer)
+    }
+    // Simply do nothing if the user has no tokens
+    catch (err) { console.log(err) }
+}
+}
```
Comment (snajima) [inline src/services/NotifService.ts:49]: same issue here

**Review by snajima — CHANGES_REQUESTED** [review]
we should set `altered_price` to the original price upon creation of a post, rather than 0 (as ur not making `altered_price` a nullable field). also, in future PRs, don't include changes for another PR in any of ur PRs.

File: src/api/controllers/NotifController.ts (hunk 4 above)
Comment (joycrwu) [reply src/api/controllers/NotifController.ts:17]: sorry i messed up the branches unfortunately, should i go and get rid of them

**Comment by joycrwu** [comment]
> we should set `altered_price` to the original price upon creation of a post, rather than 0 (as ur not making `altered_price` a nullable field). also, in future PRs, don't include changes for another PR in any of ur PRs.

i believe in the postRepository, when a new post is created i set the altered_price as the input price given from the user

**Review by snajima — APPROVED** [review]
lgtm!

**Review by mateow99 — APPROVED** [review]
LGTM

**Pull request merged**